utils/misc.ml

```diff
@@ -546,7 +546,8 @@ let spellcheck env name =
          else if dist = best_dist then (head :: best_choice, dist)
          else acc
   in
-  fst (List.fold_left (compare name) ([], max_int) env)
+  fst (List.fold_left (compare name) ([], max_int)
+         (List.sort_uniq String.compare env))
```

Comment:
Currently, the order of the elements in the output list is the reverse of the order in the `env` input list. After this patch, it is set to the reverse of the alphabetical order (as can be seen in the currently failing test).

I think it's fine for this PR to change the order (the `env` input list was coming from a fold on the environment and hence in an unspecified order anyway). I would just slightly prefer if it was the alphabetical order, not the reversed alphabetical order :-).

Reply:
Thanks, fixed.